dandiapi/api/services/metadata/__init__.py

```diff
@@ -95,6 +96,7 @@ def version_aggregate_assets_summary(version: Version) -> None:
     )
     if updated_count == 0:
         logger.info('Skipped updating assetsSummary for version %s', version.id)
+        raise VersionMetadataConcurrentlyModified
```

Comment:
Will this trigger a Sentry error? I imagine we wouldn't want that to happen in this case.

Reply:
I don't think it will. I _believe_ what happens is the following:

1. This exception is raised
2. Since the task is decorated with `autoretry_for`, it catches that exception and raises a celery `Retry` exception
3. Sentry (specific the celery integration) sees the `Retry` exception is raised, but ignores it, as it's recognized as part of celery's control flow. [Here](https://github.com/getsentry/sentry-python/blob/2f05ccbae7298978d2f1a8774a07386a018bcce9/sentry_sdk/integrations/celery.py#L346) is the code that controls that.

So we should be good, but I'll do a little more testing to make sure things behave as we expect before merging.